### PR TITLE
internal/integration: add schema diff test to postgres

### PR DESCRIPTION
**Pull request by ronenlu**

### CODE REVIEW

File: internal/integration/postgres_test.go [hunk 1]
```diff
@@ -812,6 +812,76 @@ CREATE TABLE "other"."users" ("id" integer NOT NULL);
 	})
 }
 
+func TestPostgres_SchemaDiff(t *testing.T) {
+	bin, err := buildCmd(t)
+	require.NoError(t, err)
+	pgRun(t, func(t *pgTest) {
+		dir := t.TempDir()
+		_, err = t.db.Exec("CREATE DATABASE test1")
```
Comment (rotemtam) [inline internal/integration/postgres_test.go:820]: t.Clean?

File: internal/integration/postgres_test.go [hunk 2]
```diff
@@ -812,6 +812,76 @@ CREATE TABLE "other"."users" ("id" integer NOT NULL);
...
+		require.NoError(t, err)
+		_, err = t.db.Exec("CREATE DATABASE test2")
+		require.NoError(t, err)
+		defer t.db.Exec("DROP DATABASE IF EXISTS test1")
```
Comment (masseelch) [inline internal/integration/postgres_test.go:824]: Use `t.CleanUp` instead of defer in tests

File: internal/integration/postgres_test.go [hunk 3]
```diff
@@ -812,6 +812,82 @@ CREATE TABLE "other"."users" ("id" integer NOT NULL);
...
+		diff := func(db1, db2 string) string {
+			out, err := exec.Command(
+				bin, "schema", "diff",
+				"--from", fmt.Sprintf("postgres://postgres:pass@localhost:%d/%s", t.port, db1),
+				"--to", fmt.Sprintf("postgres://postgres:pass@localhost:%d/%s", t.port, db2),
+			).CombinedOutput()
+			require.NoError(t, err, string(out))
+			return strings.TrimSpace(string(out))
+		}
+		// diff databases with same realm
```
Comment (a8m) [inline internal/integration/postgres_test.go:842]:
```suggestion
		// Diff a database with itself.
```

File: internal/integration/postgres_test.go [hunk 4]
```diff
@@ -812,6 +812,82 @@ CREATE TABLE "other"."users" ("id" integer NOT NULL);
...
+		require.Equal(t, "Schemas are synced, no changes to be made.", diff("test1?sslmode=disable", "test2?sslmode=disable"))
+
+		// create schemas on test2 DB
```
Comment (a8m) [inline internal/integration/postgres_test.go:845]:
```suggestion
		// Create schemas on test2 database.
```

File: internal/integration/postgres_test.go [hunk 5]
```diff
@@ -812,6 +812,82 @@ CREATE TABLE "other"."users" ("id" integer NOT NULL);
...
+		require.NoError(t, err)
+		out, err := exec.Command(
+			bin, "schema", "apply",
+			"-u", fmt.Sprintf("postgres://postgres:pass@localhost:%d/test2?sslmode=disable", t.port),
+			"-f", fmt.Sprintf(filepath.Join(dir, "schema.hcl")),
+			"--auto-approve",
+		).CombinedOutput()
+		require.NoError(t, err, string(out))
+
+		// diff databases with different realm
```
Comment (a8m) [inline internal/integration/postgres_test.go:868]:
```suggestion
		// Diff a database with different one.
```
Use either database or realm.

File: internal/integration/postgres_test.go [hunk 6]
```diff
@@ -812,6 +812,82 @@ CREATE TABLE "other"."users" ("id" integer NOT NULL);
...
+		require.Equal(t, `-- Add new schema named "other"
+CREATE SCHEMA "other"
+-- Create "users" table
+CREATE TABLE "public"."users" ("id" integer NOT NULL)
+-- Create "posts" table
+CREATE TABLE "other"."posts" ("id" integer NOT NULL)`, diff("test1?sslmode=disable", "test2?sslmode=disable"))
+		// diff schemas
```
Comment (a8m) [inline internal/integration/postgres_test.go:875]: These ones are already tested in: https://github.com/ariga/atlas/blob/master/internal/integration/integration_test.go#L397

**Review by a8m — APPROVED** [review]

**Pull request merged**